clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java

```diff
@@ -211,16 +212,18 @@ public Integer getLatestValueSchemaId() throws VeniceClientException {
 
   @Override
   public int getValueSchemaId(Schema schema) {
-    if (valueSchemaMapR.getIfPresent(schema) == null) {
+    String canonicalSchemaStr = AvroCompatibilityHelper.toParsingForm(schema);
```

Comment:
Hmm, should we move this path into below when original schema -> id is not found? Otherwise each schema id lookup will have to parse the schema? 
Like, the reserve mapping should be original schema -> value schema ID still..

Reply:
That does make sense. Let me think about it